### PR TITLE
Changed do...while

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

Just to avoid an error in ECG sniffs ... https://github.com/sreichel/magento-lts/actions/runs/3554950840/jobs/5971362387

### Related Issues

1. See magento-ecg/coding-standard#70

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php [hunk 1]
```diff
@@ -165,15 +165,15 @@ public function getMinValue()
      */
     public function getRange()
     {
-        $range = $this->getData('range');
+        $range = (float)$this->getData('range');
         if (!$range) {
             $maxValue = $this->getMaxValue();
             $index = 1;
-            do {
+            while (!isset($items) || $range > self::MIN_RANGE_POWER && count($items) < 2) {
```
Comment (Flyingmana) [inline app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php:172]: changing this to a `while` from a `do...while` makes it harder to understand. Is this rule of the coding standard important enough so we have to follow it?

File: app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php [hunk 2]
```diff
@@ -184,7 +184,7 @@ public function getRange()
      * Retrieve information about products count in range
      *
      * @param int $range
-     * @return int
+     * @return array
```
Comment (Flyingmana) [inline app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php:187]: wrongful (and unprecise) change

File: app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php:172]: No, Thats can be closed. Already opened issue for that.

File: app/code/core/Mage/Core/Model/Config.php [hunk 3]
```diff
@@ -483,15 +483,15 @@ protected function _canUseCacheForInit()
     {
         if (Mage::app()->useCache('config') && $this->_allowCacheForInit) {
             $retries = 10;
-            do {
+            while ($retries--) {
```
Comment (Flyingmana) [inline app/code/core/Mage/Core/Model/Config.php:486]: this reduces the retries by one compared to before the change

File: app/code/core/Mage/Persistent/Model/Session.php [hunk 4]
```diff
@@ -120,9 +120,9 @@ protected function _beforeSave()
         if ($this->isObjectNew()) {
             $this->setWebsiteId(Mage::app()->getStore()->getWebsiteId());
             // Setting cookie key
-            do {
+            while (!$this->getResource()->isKeyAllowed($this->getKey())) {
```
Comment (Flyingmana) [inline app/code/core/Mage/Persistent/Model/Session.php:123]: was it validated, that this is working as expected on the first run, when no key was yet set?

**Review by Flyingmana — CHANGES_REQUESTED** [review]

File: app/code/core/Mage/Core/Model/Config.php (hunk 3 above)
Comment (sreichel) [reply app/code/core/Mage/Core/Model/Config.php:486]: this was not complete and marked as draft

File: app/code/core/Mage/Persistent/Model/Session.php (hunk 4 above)
Comment (sreichel) [reply app/code/core/Mage/Persistent/Model/Session.php:123]: Did not commit latest changes and opend issue instead ... https://github.com/magento-ecg/coding-standard/issues/70

File: app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Layer/Filter/Decimal.php:187]: This one is correct ... see `Mage_Catalog_Model_Resource_Layer_Filter_Decimal::getCount()`

**Pull request closed without merging**